x-pack/platform/plugins/shared/security/server/authentication/providers/token.ts

```diff
@@ -210,23 +210,21 @@ export class TokenAuthenticationProvider extends BaseAuthenticationProvider {
     try {
       refreshTokenResult = await this.options.tokens.refresh(state.refreshToken);
     } catch (err) {
-      this.logger.error(`Failed to refresh access token: ${getDetailedErrorMessage(err)}`);
-      return AuthenticationResult.failed(err);
-    }
-
-    // If refresh token is no longer valid, then we should clear session and redirect user to the
-    // login page to re-authenticate, or fail if redirect isn't possible.
-    if (refreshTokenResult === null) {
-      if (canStartNewSession(request)) {
-        this.logger.warn('Clearing session since both access and refresh tokens are expired.');
+      // If refresh token is no longer valid, then we should clear session and redirect user to the
+      // login page to re-authenticate, or fail if redirect isn't possible.
+      if (err instanceof InvalidGrantError) {
+        if (canStartNewSession(request)) {
+          this.logger.warn('Clearing session since both access and refresh tokens are expired.');
```

Comment:
Should we tweak this log message? Or is the original still valid?

```suggestion
          this.logger.warn('Clearing session since refresh token is expired.');
```